app/javascript/components/top-navbar/right-section.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import CustomLogo from './custom-logo';
+import Notifications from './notifications';
+import Help from './help';
+import Configuration from './configuration';
+import UserOptions from './user-options';
+import {
+  helpMenuProps, recursiveHelpMenuProps, userMenuProps, recursiveUserMenuProps,
+} from './recursive-props';
+
+const RightSection = ({
+  customLogo, currentUser, helpMenu, opsExplorerAllowed, applianceName, miqGroups, currentGroup, userMenu,
+}) => (
+  <React.Fragment>
+    <CustomLogo customLogo={customLogo} />
+    <Notifications />
+    <Help helpMenu={helpMenu} />
+    { currentUser && (
+      <React.Fragment>
+        <Configuration opsExplorerAllowed={opsExplorerAllowed} />
```

Comment:
TBH I don't see a reason why to have this as a separate component in a separate file, this is way too small for that and you're creating much bigger ones here. I'd just extract it into a function, but don't take this seriously and ask someone who actually knows React :wink:

Reply:
This was actually by my command! :skull_and_crossbones: 

But it is better to make these component as small and independent as possible. It is much easier to find them, test them and fix them if need be.

Reply:
Then IMO the other one(s) should be small(er) as well... It's just feels weird that you have one small, and the rest is HUGE.